sylvia/tests/reply_dispatch.rs

```diff
@@ -0,0 +1,371 @@
+}
+
+pub struct Contract<M, Q> {
+    remote: Item<Remote<'static, noop_contract::NoopContract<Empty, Empty>>>,
+    last_reply: Item<u64>,
+    _phantom: std::marker::PhantomData<(M, Q)>,
+}
+
+#[entry_points(generics<Empty, Empty>)]
+#[contract]
+#[sv::error(ContractError)]
+#[sv::custom(msg=M, query=Q)]
+#[sv::features(replies)]
+impl<M, Q> Contract<M, Q>
+where
+    M: CustomMsg + 'static,
+    Q: CustomQuery + 'static,
+{
+    pub fn new() -> Self {
+        Self {
+            remote: Item::new("remote"),
+            last_reply: Item::new("last_reply"),
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    #[sv::msg(instantiate)]
+    pub fn instantiate(
+        &self,
+        ctx: InstantiateCtx<Q>,
+        remote_code_id: u64,
+    ) -> Result<Response<M>, ContractError> {
+        // Custom type can be used as a payload.
+        let payload = InstantiatePayload {
+            sender: ctx.info.sender,
+        };
+        let sub_msg = InstantiateBuilder::noop_contract(remote_code_id)?
+            .with_label("noop")
+            .build()
+            .remote_instantiated(to_json_binary(&payload)?)?;
+        // TODO: Blocked by https://github.com/CosmWasm/cw-multi-test/pull/216. Uncomment when new
+        // MultiTest version is released.
+        // Payload is not currently forwarded in the MultiTest.
+        // .remote_instantiated(payload)?;
+
+        Ok(Response::new().add_submessage(sub_msg))
+    }
+
+    #[sv::msg(exec)]
+    pub fn call_remote_success(
+        &self,
+        ctx: ExecCtx<Q>,
+        should_fail: bool,
+    ) -> Result<Response<M>, ContractError> {
+        let msg = self
+            .remote
+            .load(ctx.deps.storage)?
+            .executor()
+            .noop(should_fail)?
+            .build()
+            .success(Binary::default())?;
+
+        Ok(Response::new().add_submessage(msg))
+    }
+
+    #[sv::msg(exec)]
+    pub fn call_remote_error(
+        &self,
+        ctx: ExecCtx<Q>,
+        should_fail: bool,
+    ) -> Result<Response<M>, ContractError> {
+        let msg = self
+            .remote
+            .load(ctx.deps.storage)?
+            .executor()
+            .noop(should_fail)?
+            .build()
+            .error(Binary::default())?;
+
+        Ok(Response::new().add_submessage(msg))
+    }
+
+    #[sv::msg(exec)]
+    pub fn call_remote_both(
+        &self,
+        ctx: ExecCtx<Q>,
+        should_fail: bool,
+    ) -> Result<Response<M>, ContractError> {
+        let msg = self
+            .remote
+            .load(ctx.deps.storage)?
+            .executor()
+            .noop(should_fail)?
+            .build()
+            .both(Binary::default())?;
+
+        Ok(Response::new().add_submessage(msg))
+    }
+
+    #[sv::msg(exec)]
+    pub fn call_remote_always(
+        &self,
+        ctx: ExecCtx<Q>,
+        should_fail: bool,
+    ) -> Result<Response<M>, ContractError> {
+        // Tuple can be used as a payload.
```

Comment:
That's a nice feature. Let's describe that fact somewhere in the docs and we could remove this comment from here

Reply:
I have to check if it's well described in the code docs. I will also document it in our docs.

I would leave the comment for future developers that could consider this method redundant.